American Whitewater/Helpers/DefaultsManager.swift

```diff
@@ -1,291 +1,189 @@
     }
 
-    
-    private static let distanceFilterKey = "distanceFilterKey"
-    static var distanceFilter: Double {
+    var distanceFilter: Double {
         get {
-            var distanceFilter = UserDefaults.standard.double(forKey: distanceFilterKey)
+            var distanceFilter = defaults.double(forKey: Keys.distanceFilter)
             if distanceFilter == 0.0 {
                 distanceFilter = 100
-                UserDefaults.standard.set(distanceFilter, forKey: distanceFilterKey)
+                defaults.set(distanceFilter, forKey: Keys.distanceFilter)
             }
             
-            return UserDefaults.standard.double(forKey: distanceFilterKey)
+            return defaults.double(forKey: Keys.distanceFilter)
         }
         set {
-            UserDefaults.standard.set(newValue, forKey: distanceFilterKey)
+            defaults.set(newValue, forKey: Keys.distanceFilter)
         }
     }
     
-    private static let showDistanceFilterKey = "showDistanceFilterKey"
-    static var showDistanceFilter: Bool {
-        get {
-            return UserDefaults.standard.bool(forKey: showDistanceFilterKey)
-        }
-        
-        set {
-            UserDefaults.standard.set(newValue, forKey: showDistanceFilterKey)
-        }
+    var showDistanceFilter: Bool {
+        get { defaults.bool(forKey: Keys.showDistanceFilter) }
+        set { defaults.set(newValue, forKey: Keys.showDistanceFilter) }
     }
 
-    private static let showRegionFilterKey = "showRegionFilterKey"
-    static var showRegionFilter: Bool {
-        get {
-            return UserDefaults.standard.bool(forKey: showRegionFilterKey)
-        }
-        
-        set {
-            UserDefaults.standard.set(newValue, forKey: showRegionFilterKey)
-        }
+    var showRegionFilter: Bool {
+        get { defaults.bool(forKey: Keys.showRegionFilter) }
+        set { defaults.set(newValue, forKey: Keys.showRegionFilter) }
     }
-
     
-    private static let latitudeKey = "latitudeKey"
-    private static let longitudeKey = "longitudeKey"
-    static var latitude: Double {
-        get {
-            return UserDefaults.standard.double(forKey: latitudeKey)
-        }
-        set {
-            UserDefaults.standard.set(newValue, forKey: latitudeKey)
-        }
+    //
+    // MARK: - Location
+    //
+    
+    /// Get the last saved location as a CLLocation for convenience (since that's generally what's wanted)
+    /// this is get only to make it clear that we're not saving any of the other stuff on CLLocation (altitude, heading, etc)
+    /// set coordinate instead
+    public var location: CLLocation {
+        .init(latitude: latitude, longitude: longitude)
     }
-    static var longitude: Double {
-        get {
-            return UserDefaults.standard.double(forKey: longitudeKey)
-        }
+    
+    public var coordinate: CLLocationCoordinate2D {
+        get { .init(latitude: latitude, longitude: longitude) }
```

Comment:
What does `.init` do?

Reply:
Oh that's swift shorthand where the type can be inferred (it can here because it's declared on the previous line). So in this case, this is equivalent to:

`CLLocationCoordinate2D(latitude: latitude, longitude: longitude)`